SimCalorimetry/HGCalSimProducers/plugins/alpaka/HGCalDigiSoaFiller.cc

```diff
@@ -0,0 +1,185 @@
+    hgcaldigi::HGCalDigiHost host_buffer(finaldigi_size, cms::alpakatools::host());
+    int nmatch(0), nfail(0);
+    for (auto d : digisCEE) {
+      //det id to electronics id
+      uint32_t detid(d.id().rawId());
+      HGCalElectronicsId eleid(
+          ::hgcal::mappingtools::getElectronicsIdForSiCell<ModuleInfo, CellInfo>(modules, cells, detid));
+      if (eleid.raw() == 0) {
+        nfail++;
+        continue;
+      }
+
+      //get the index in the SOA
+      auto modcellidx =
+          ::hgcal::mappingtools::getModuleCellIndicesForSiCell<ModuleInfo, CellInfo>(modules, cells, detid);
+      uint32_t i = finaldigi_size;
+      try {
+        i = modidx.getIndexForModuleData(
+            eleid.localFEDId(), eleid.captureBlock(), eleid.econdIdx(), eleid.econdeRx(), eleid.halfrocChannel());
+        if (i >= finaldigi_size) {
+          HGCSiliconDetId siid(d.id());
+          int32_t midx = modcellidx.first;
+          int32_t cidx = modcellidx.second;
+          std::cout << "Failed to get proper index (" << i << ") for "
+                    << "layer=" << siid.layer() << " u=" << siid.waferU() << " v=" << siid.waferV() << std::endl
+                    << "\t Modidx:" << midx << " cellidx: " << cidx << " cellu=" << siid.cellU()
+                    << " cellv=" << siid.cellV() << std::endl
+                    << "\tModule info FED:" << (uint32_t)modules.view()[midx].fedid()
+                    << " CB:" << (uint32_t)modules.view()[midx].captureblockidx()
+                    << " ECOND: " << (uint32_t)modules.view()[midx].econdidx()
+                    << "\tEle id: FED=" << (uint32_t)eleid.localFEDId() << " CB:" << (uint32_t)eleid.captureBlock()
+                    << " ECONDidx: " << (uint32_t)eleid.econdIdx() << std::endl;
+          nfail++;
+          continue;
+        }
+      } catch (std::exception& e) {
+        std::cout << e.what() << std::endl;
+        nfail++;
+        continue;
+      }
+
+      nmatch++;
+
+      //read digi (in-time sample only)
+      uint32_t toa = d.sample(itSample).toa();
+      uint32_t tctp = d.sample(itSample).mode() ? 3 : 0;
+      uint32_t adcm1(d.sample(itSample - 1).data());
+      uint32_t adc(tctp < 3 ? d.sample(itSample).data() : 0);
+      uint32_t tot(tctp == 3 ? d.sample(itSample).data() : 0);
+
+      //fill SOA
+      auto idigi = host_buffer.view()[i];
+      //idigi.electronicsId() = eleid.raw();
+      idigi.tctp() = tctp;
+      idigi.adcm1() = adcm1;
+      idigi.adc() = adc;
+      idigi.tot() = tot;
+      idigi.toa() = toa;
+      idigi.cm() = 0;
+      idigi.flags() = 0;
+    }
+
+    std::cout << "Matched: " << nmatch << " failed: " << nfail << std::endl;
+
+    //allocate device colleciton, copy from host and put in event
+    auto& queue = iEvent.queue();
+    hgcaldigi::HGCalDigiDevice device_buffer(host_buffer.view().metadata().size(), queue);
+    alpaka::memcpy(queue, host_buffer.buffer(), device_buffer.const_buffer());
```

Comment:
If this is supposed to be a host-to-device copy (as suggested by the comment), it should be
```suggestion
    alpaka::memcpy(queue, device_buffer.buffer(), host_buffer.const_buffer());
```

Reply:
This file was remove all together in https://github.com/CMS-HGCAL/cmssw/pull/137.